packages/builder-util-runtime/src/httpExecutor.ts

```diff
@@ -485,9 +485,8 @@ function configurePipes(options: DownloadCallOptions, response: IncomingMessage)
   let lastStream = response
   for (const stream of streams) {
     stream.on("error", (error: Error) => {
-      if (!options.options.cancellationToken.cancelled) {
-        options.callback(error)
-      }
+      fileOut.close()
+      options.callback(error)
```

Comment:
Great callout on closing the stream

I think we still need: `if (!options.options.cancellationToken.cancelled) {` though

Reply:
callback should be invoked even if the downloading is cancelled by user.

Reply:
> Great callout on closing the stream
> 
> I think we still need this though:
> 
> ```
> if (!options.options.cancellationToken.cancelled) {
>        options.callback(error)
> }
> ```

callback should be invoked even if the downloading is cancelled by user.

Reply:
Hmm, I disagree though, as user cancelation is an explicit action that I think should not resolve into an error event.

Reply:
agree!~